rosidl_typesupport_introspection_c/cmake/rosidl_typesupport_introspection_c_generate_interfaces.cmake

```diff
@@ -23,15 +25,23 @@ foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
     get_filename_component(_parent_folder "${_parent_folder}" NAME)
     get_filename_component(_msg_name "${_idl_file}" NAME_WE)
     string_camel_case_to_lower_case_underscore("${_msg_name}" _header_name)
-    list(APPEND _generated_header_files
+    list(APPEND _generated_msg_header_files
       "${_output_path}/${_parent_folder}/${_header_name}__introspection_type_support.h"
     )
-    list(APPEND _generated_source_files
+    list(APPEND _generated_msg_source_files
       "${_output_path}/${_parent_folder}/${_header_name}__type_support.c"
     )
   elseif("${_extension} " STREQUAL ".srv ")
-    # no generated code for services
-    # only for the request / response messages
+    get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
+    get_filename_component(_parent_folder "${_parent_folder}" NAME)
+    get_filename_component(_msg_name "${_idl_file}" NAME_WE)
+    string_camel_case_to_lower_case_underscore("${_msg_name}" _header_name)
```

Comment:
The lines 35-38 are identical to the previous block. I would suggest to move them above the conditions in order to deduplicate.